### PR TITLE
Ensure pprof executes commands once per interval

### PR DESCRIPTION
**Pull request by portante**
Fixes issue #871.

### CODE REVIEW

**Comment by sjug** [comment]
I've tried this out and it seems to break pprof collection in the sense that no samples are collected, the result directory is empty and no pprof commands are executed.

**Comment by portante** [comment]
@sjug, thanks, I refactored the code in newly added commit.  Let's see if that helps us debug what is going on.  Removing redundancy is a good place to start when debugging a problem.

File: agent/tool-scripts/pprof [hunk 1]
```diff
@@ -100,77 +99,121 @@ done
...
+    suffix=$4
+    oc get --raw /debug/pprof/heap --server ${server} > ${pbench_tmp}/${target}${suffix}_heap && tool ${tool} -text -seconds=${interval} -output ${PPROF_TMPDIR}/${target}/${datestamp}${suffix}_heap.txt ${pbench_tmp}/${target}${suffix}_heap
+}
+
+_pprof_profile() {
+    datestamp=$1
+    server=$2
+    target=$3
+    suffix=$4
+    oc get --raw /debug/pprof/profile?seconds=${interval} --server ${server} > ${pbench_tmp}/${target}${suffix}_profile && tool ${tool} -text -seconds=${interval} -output ${PPROF_TMPDIR}/${target}/${datestamp}${suffix}_profile.txt ${pbench_tmp}/${target}${suffix}_profile
```
Comment (sjug) [inline agent/tool-scripts/pprof]: `${tool_bin} tool ${tool}`

**Comment by chaitanyaenr** [comment]
Verified that the tool is working as expected.

**Pull request merged**